builtin/logical/aws/backend_test.go

```diff
@@ -655,17 +658,17 @@ func testAccStepRead(t *testing.T, path, name string, credentialTests []credenti
 	}
 }
 
-func testAccStepReadTTL(name string, maximumTTL time.Duration) logicaltest.TestStep {
+func testAccStepReadSTSResponse(name string, maximumTTL uint64) logicaltest.TestStep {
 	return logicaltest.TestStep{
 		Operation: logical.ReadOperation,
 		Path:      "creds/" + name,
 		Check: func(resp *logical.Response) error {
-			if resp.Secret == nil {
-				return fmt.Errorf("bad: nil Secret returned")
+			if resp.Secret != nil {
+				return fmt.Errorf("bad: STS tokens should return a nil secret, received: %+v", resp.Secret)
 			}
-			ttl := resp.Secret.TTL
+			ttl := resp.Data["ttl"].(uint64)
```

Comment:
We could make this fail a little more gracefully (as opposed to `panic`'ing) by checking that "ttl" exists before coercing it to `uint64`.

Reply:
Good call, I'll add that in!